src/lib/core/CHIPConfig.h

```diff
@@ -187,6 +187,18 @@
 #define CHIP_CONFIG_GLOBALS_NO_DESTRUCT 0
 #endif // CHIP_CONFIG_GLOBALS_NO_DESTRUCT
 
+/**
+ *  @def CHIP_CONFIG_DYNAMICALLY_ALLOCATED_GLOBALS
+ *
+ *  @brief
+ *    Whether to use dynamic allocation for chip::Global objects.
+ *
+ *    The default is to use static allocation.
+ */
+#ifndef CHIP_CONFIG_DYNAMICALLY_ALLOCATED_GLOBALS
```

Comment:
Can we call rename this to `CHIP_CONFIG_GLOBALS_DYNAMIC`, or something that starts with `CHIP_CONFIG_GLOBALS_` anyway?

This also semantically implies `CHIP_CONFIG_GLOBALS_LAZY_INIT` and/or conflicts with that setting if we take `CHIP_CONFIG_GLOBALS_LAZY_INIT` to mean "static but lazy". Maybe the simplest thing is to just check if `CHIP_CONFIG_GLOBALS_DYNAMIC` and `CHIP_CONFIG_GLOBALS_LAZY_INIT` are both 1 and `#error` if so.